web/src/main/java/se/inera/intyg/webcert/web/service/unansweredcommunication/UnansweredCommunicationServiceImpl.java

```diff
@@ -0,0 +1,127 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * sklintyg is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package se.inera.intyg.webcert.web.service.unansweredcommunication;
+
+import static se.inera.intyg.clinicalprocess.healthcond.certificate.getcertificateadditions.v1.StatusType.BESVARAD;
+import static se.inera.intyg.clinicalprocess.healthcond.certificate.getcertificateadditions.v1.StatusType.OBESVARAD;
+import static se.inera.intyg.webcert.persistence.arende.model.ArendeAmne.KOMPLT;
+
+import java.time.LocalDateTime;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Service;
+import se.inera.intyg.clinicalprocess.healthcond.certificate.getcertificateadditions.v1.StatusType;
+import se.inera.intyg.webcert.persistence.arende.model.Arende;
+import se.inera.intyg.webcert.persistence.arende.model.ArendeAmne;
+import se.inera.intyg.webcert.persistence.model.Status;
+import se.inera.intyg.webcert.web.service.arende.ArendeService;
+import se.inera.intyg.webcert.web.web.controller.internalapi.dto.UnansweredCommunicationRequest;
+import se.inera.intyg.webcert.web.web.controller.internalapi.dto.UnansweredCommunicationResponse;
+import se.inera.intyg.webcert.web.web.controller.internalapi.dto.UnansweredQAs;
+
+@Service
+public class UnansweredCommunicationServiceImpl implements UnansweredCommunicationService {
+
+    private final ArendeService arendeService;
+    private static final String FK = "FK";
+
+    public UnansweredCommunicationServiceImpl(ArendeService arendeService) {
+        this.arendeService = arendeService;
+    }
+
+    @Override
+    public UnansweredCommunicationResponse get(UnansweredCommunicationRequest request) {
+        final var maxDaysOfUnansweredCommunication = request.getMaxDaysOfUnansweredCommunication();
+        final var patientIds = request.getPatientIds();
+
+        if (patientIds == null || patientIds.isEmpty() || maxDaysOfUnansweredCommunication == null) {
+            throw new IllegalArgumentException(String.format(
+                "Request is missing parameters required to make request. patientIds '%s' maxDaysOfUnansweredCommunication '%s'",
+                patientIds,
+                maxDaysOfUnansweredCommunication));
+        }
+
+        final var arendenForPatients = arendeService.getArendenForPatientsWithTimestampAfterDate(
+            formatPatientIds(patientIds),
+            LocalDateTime.now().minusDays(maxDaysOfUnansweredCommunication)
```

Comment:
LocalDateTime.now() will return current time. For example if the query is done at 14:00 only "Arenden" after that timestamp on the correct day will be included. Set the time to the start of the day.

Reply:
Changed it to the following ```LocalDateTime.now().minusDays(maxDaysOfUnansweredCommunication).toLocalDate().atStartOfDay()```